server/xcode/xcode_darwin.go

```diff
@@ -126,13 +140,51 @@ func (x *xcodeLocator) locate() error {
 				continue
 			}
 			log.Debugf("Mapped Xcode Version %s=>%s with SDKs %+v", version, developerDirPath, sdks)
-			versionMap[version] = &xcodeVersion{
+			xv := &xcodeVersion{
 				version:          mostPreciseVersion,
 				developerDirPath: developerDirPath,
 				sdks:             sdks,
 			}
+			versionMap[version] = xv
+			if defaultDeveloperDir == developerDirPath {
+				versionMap[""] = xv
+			}
 		}
 	}
+
+	if xv, ok := versionMap[""]; ok {
+		log.Infof("xcode-select set to Xcode version %s", xv.version)
+	} else {
```

Comment:
nit: can we return early here and unnest the else? (maybe unnecessary after the other cleanups)

Reply:
Addressed in #1354.